app/models/manageiq/providers/amazon/cloud_manager/metrics_capture.rb

```diff
@@ -145,38 +154,47 @@ def counter_values_by_timestamp(metrics_by_counter_name)
         value   = i[:calculation].call(*metrics, interval)
 
         # For (temporary) symmetry with VIM API we create 20-second intervals.
+
         (last_ts + 20.seconds..ts).step_value(20.seconds).each do |inner_ts|
           counter_values_by_ts.store_path(inner_ts.iso8601, i[:vim_style_counter_key], value)
         end
       end
     end
+
     counter_values_by_ts
   end
 
   def metrics_by_counter_name(cloud_watch, counters, start_time, end_time)
     metrics_by_counter_name = {}
+
     counters.each do |c|
-      metrics = metrics_by_counter_name[c.metric_name] = {}
+      metrics = metrics_by_counter_name[c.metric_name] ||= {}
+      options = c.to_hash.merge(:statistics => %w[Average].freeze, :period => 60).freeze
```

Comment:
Is this refactoring?  It looks like all the same keys are all still here

Reply:
Yes, just refactoring, I took repeated steps out of the cycle below.